server/src/main.rs

```diff
@@ -1,53 +1,69 @@
+mod config;
 mod game;
 mod routes;
 
 use std::io;
 
 use actix::Actor;
 use actix_files::Files;
-use actix_session::{storage::CookieSessionStore, SessionMiddleware};
+use actix_session::storage::{CookieSessionStore, SessionStore};
+use actix_session::SessionMiddleware;
 use actix_web::{cookie::Key, middleware::Logger, web, App, HttpServer};
+use clap::Parser;
+use log::LevelFilter;
 
+use crate::config::Config;
 use crate::game::server::WsServer;
 
-/// Returns a secret key that is used to encrypt the session cookie stored by the client.
-///
-/// # Panics
-///
-/// Panics if the environment variable `SESSION_SECRET` is not a base64-encoded string of at least
-/// 32 randomly-generated bytes.
-fn get_session_secret() -> Key {
-    let session_secret = dotenv::var("SESSION_SECRET")
-        .expect("environment variable `SESSION_SECRET` must be defined");
-    let session_secret = base64::decode(session_secret)
-        .expect("cannot decode environment variable `SESSION_SECRET` from base64");
-    Key::derive_from(&session_secret)
+/// Command-line arguments.
+#[derive(Parser)]
+#[clap(version, author, about, long_about = None)]
+struct Args {
+    /// The port to which the web server is bound.
+    #[clap(short, long, default_value_t = 3000)]
+    port: u16,
+    /// The verbosity level of the log.
+    #[clap(short, long, default_value_t = LevelFilter::Info)]
+    verbosity: LevelFilter,
+}
+
+/// Builds a middleware that manages sessions.
+fn build_session_middleware(config: &Config) -> SessionMiddleware<impl SessionStore> {
+    let session_secret =
+        base64::decode(&config.session_secret).expect("cannot decode session secret from Base64");
+
+    SessionMiddleware::builder(
+        CookieSessionStore::default(),
+        Key::derive_from(&session_secret),
+    )
+    .cookie_name("session".to_string())
+    .build()
 }
 
 /// Launches the web server.
 #[actix_web::main]
 async fn main() -> io::Result<()> {
-    env_logger::init();
+    let args = Args::parse();
 
-    let game_server = WsServer::default().start();
+    env_logger::builder().filter_level(args.verbosity).init();
+
+    let config = Config::new().expect("cannot read configuration");
+
+    let game_server = WsServer::new().start();
```

Comment:
Why swap back to `new()` from `default()`?

Reply:
Just based on what I've seen in popular crates/other Rust code. I think people tend to use `new` when manually making a new instance of a struct, whereas `default` is used when trait constraints require a type that is `Default`, and are unaware of a conventional-but-not-guaranteed `new` method.